k2/csrc/host_shim.h

```diff
@@ -179,6 +182,163 @@ class FsaVecCreator {
   int32_t next_fsa_idx_;
 };
 
+/*
+  Wrap some property test function in `host/properties.h`, they are generally
+  for test purpose for now and work only for CPU. Users usually would not call
+  these functions. Instead, Call `GetFsaBasicProperties` or
+  `GetFsaVecBasicProperties` in fsa.h if you want to check Fsa/FsaVec's
+  properites in production code.
+*/
+
+/*
+  Check properties of FsaOrVec (which must be on CPU) with property test
+  function `f` which is one of property test functions for k2host::Fsa in
+  host/properties.h, e.g. IsValid(const k2::host Fsa&), IsTopSorted(const
+  k2host::Fsa&).
+
+  If `fsas` is FsaVec, the function will return an array on CPU which has
+    ans[i] = f(FsaVecToHostFsa(fsa_vec, i)) for 0 <= i < fsa_vec.Dim0();
+  else
+    returns a CPU array with size 1 and ans[0] = f(FsaToHostFsa(fsas))
+*/
+Array1<bool> CheckProperties(FsaOrVec &fsas, bool (*f)(const k2host::Fsa &));
```

Comment:
I would suggest moving it to host_shim.cc and giving it a local linkage.
Users will not invoke it directly.

Reply:
Sure, Will do